spring-kafka-avro/docker-compose.yml

```diff
@@ -0,0 +1,22 @@
+version: '3'
+services:
+  kafka:
+    image: wurstmeister/kafka
+    hostname: kafka
+    ports:
+      - "9092:9092"
```

Comment:
The other docker-compose scripts use the single port notation, which will prevent port conflicts. Can this be done here, too?

Reply:
Unfortunately no; the kafka clients connect to the bootstrap server(s) which then return the actual "advertised hosts/ports" to connect to.

The [README}(https://github.com/wurstmeister/kafka-docker/blob/master/README.md#advertised-port) talks about a mechanism to dynamically set the advertised hosts property, but I couldn't get it to work.

I exec'd into the container and tried to run the suggested command but I got some docker error.

My docker chops are not great; I am open to suggestions.

Reply:
I figured it out...